site/soc/software/charts/osh/openstack-tempest/tempest.yaml

```diff
@@ -29,7 +29,10 @@ data:
         -w {{ tempest_workers }} \
         {{ tempest_test_args[tempest_test_type] }} \
 {% if use_blacklist %}
-        --blacklist-file /etc/tempest/test-blacklist
+        --blacklist-file /etc/tempest/test-blacklist \
+{% endif %}
+        || true
```

Comment:
Im wondering, if we use the blacklist to skip those failed tests, we dont need to exit 0 because the tempest run will work as expected so can we make this a if-else? either we are using the blacklist and we expect the exit code to be the real code or we dont and then we exit 0 as to not block the depoloyment.

I think this is a bad idea still, we should make sure that we blacklist those failing tests that we cannot do anything about *always* and let tempest exit with the real status because if not, we are hiding any problems. We cannot consider deploying tempest if we are gonna ignore the actual state of the cloud tests.

Reply:
Maybe this is me butting heads with the design strategy then. To me, a failed tempest test should not equal a failed pod / failed release. The tempest test pod has (essentially) a single job to do: run all the tempest tests defined for that run, and that's it. If it encounters problems that prevent it from running tests (can't connect to host, can't locate config file, can't write to file system, etc.) then that's a legitimate error and the pod should be moved to 'Error' state because it couldn't complete its job and the tests didn't run. But if all tests were run to completion, then the pod has accomplished its job and it makes sense to me to move it to 'Completed' state. 

If individual tempest tests fail, that shouldn't mean the deployment of the tempest release has failed. A failed test here seems like expected, or at least reasonable, behavior. This might be getting too abstract, but if I think of a group of kids taking an exam, no one would claim the exam didn't happen or the examination process had failed if a couple of kids failed their tests. Maybe if the exam room was locked, the proctor didn't show up, there was a fire drill, etc., then sure, but a couple kids failing their tests is an expected outcome that's independent of the success or failure of the exam process itself.

Okay, I'll shut up now. I definitely see your point, but I wish the state of the pod was totally separate from the state of the individual tests.

Reply:
Ummm I see, that is a good point. Maybe Im just using the other pods as example, like when launching neutron if neutron fails, pod fails, but indeed this may be a different case as technically tempest has run as expected....

Ok, seems good, Im convinced :+1: , I guess we should also tackle this upstream somewhow...